core/store/src/trie/iterator.rs

```diff
@@ -124,8 +122,10 @@ impl<'a> TrieIterator<'a> {
         Ok(hash)
     }
 
-    fn descend_into_node(&mut self, node: TrieNodeWithSize) {
+    fn descend_into_node(&mut self, hash: &CryptoHash) -> Result<(), StorageError> {
```

Comment:
add comment (from what I understand this is only used to mark the trail, right?)